src/Shared/RangeHelper/RangeHelper.cs

```diff
@@ -107,7 +109,7 @@ internal static RangeItemHeaderValue NormalizeRange(RangeItemHeaderValue range,
                     end = length - 1;
                 }
             }
-            else
+            else if (end.HasValue)
```

Comment:
I couldn't figure out if there was a guarantee that when start is null, end is not null (or vice-versa).